packages/mobile/src/components/details-tile/DetailsTile.tsx

```diff
@@ -222,14 +226,20 @@ export const DetailsTile = ({
   }, [onPressPlay])
 
   const renderCornerTag = () => {
-    if (isPremiumContentEnabled && premiumConditions) {
-      return (
-        <PremiumTrackCornerTag
-          doesUserHaveAccess={doesUserHaveAccess}
-          isOwner={isOwner}
-          premiumConditions={premiumConditions}
-        />
-      )
+    const showPremiumCornerTag =
+      isPremiumContentEnabled &&
+      premiumConditions &&
+      (isOwner || !doesUserHaveAccess)
+    const cornerTagIconType = showPremiumCornerTag
+      ? isOwner
```

Comment:
Just making sure here - does this logic translate to:
if isOwner, show lock icon
else show nothing
Or am I misreading this?

Reply:
Ah no I think I'm wrong - if isOwner, enter the next branch, else show lock.

Reply:
it translates to:
- if owner, then see the premium conditions
  - if collectible gated, then show collectible gated that icon
  - if special access, then show special access icon
- if not owner, then show locked icon